File: search/[boj]1260_dfs와bfs_탐색.cpp
#include <iostream>
#include <queue>
#include <map>
#include <set>

using namespace std;

map<int,set<int>> graph;
int visitedL[1001];

void dfs(int v) {
	cout << v << " ";
	visitedL[v] = 1;
	for (auto it = graph[v].begin(); it != graph[v].end(); it++) {
		if (!visitedL[*it]) {
			dfs(*it);
		}
	}
}

void bfs(int v) {
	queue<int> q;
	q.push(v);
	visitedL[v] = 1;
	int temp;
	while (!q.empty()) {
		temp = q.front(); q.pop();
		cout << temp << " ";
		for (auto it = graph[temp].begin(); it != graph[temp].end(); it++) {

			if (!visitedL[*it]) {
				q.push(*it);
				visitedL[*it] = 1;
			}
		}
	}
}

int main() {
	int N, M, V;
	cin >> N >> M >> V;
	int temp1, temp2;
	for (int i = 0; i < M; i++) {
		cin >> temp1 >> temp2;
		if (graph.find(temp1) == graph.end()) {
			graph[temp1] = { temp2 };
		}
		else {
			graph[temp1].insert(temp2);
		}

		if (graph.find(temp2) == graph.end()) {
			graph[temp2] = { temp1 };
		}
		else {
			graph[temp2].insert(temp1);
		}
	}

	dfs(V);
	for (int i = 0; i < 1001; i++) {
		visitedL[i] = 0;
	}
	cout << endl;
	bfs(V);

	return 0;
}

File: search/[boj]1260_dfs와bfs_탐색.py
import queue
import sys
sys.setrecursionlimit(10**6)

N,M,V=map(int,input().split())
graph={}

for i in range(M):
    a,b=map(int,input().split())
    if a in graph:
        graph[a].append(b)
    else:
        graph[a]=[b]
    if b in graph:
        graph[b].append(a)
    else:
        graph[b]=[a]
# 작은수부터 나오게 sorting
for i in graph:
    graph[i].sort()

def dfs(V):
    print(V,end=" ")
    visitedL[V]=1
    if V in graph:
        for node in graph[V]:
            if not visitedL[node]:
                dfs(node)

def bfs(V):
    q=queue.Queue()
    q.put(V)
    visitedL[V] = 1
    while(not q.empty()):
        temp=q.get()
        print(temp,end=" ")
        if temp in graph:
            for node in graph[temp]:
                if not visitedL[node]:
                    q.put(node)
                    visitedL[node]=1



visitedL=[0 for _ in range(N+1)]
dfs(V)
print()

visitedL=[0 for _ in range(N+1)]
bfs(V)
print()